### DISCUSSION TITLE
How to properly use DateTime with Microsoft Sql Server?

### DISCUSSION THREAD

**Discussion by rstoye** [prisma/prisma#22011] [Q&A] [answered]
### Question

I'm a bit perplexed with this and I'm starting to question my sanity.

I have an old database that is used for several old applications. Currently I'm tasked with setting up a modern application to access some data in this old database. I'm using next.js and typescript for that. There are no plans to change anything in the structure of the database, it's tables or any fields. Therefore I can't actually use the prisma migration feature.

The tables I want to access all have a field called `BEARBEITET_AM`, which is German for "changedOn" or "changedAt". The field is a datetime and has the default value `GETDATE()` which is synonymous with `CURRENT_TIMESTAMP` and returns the current server date and time. In our case the server date and time is set to UTC+1.

No matter how I declare the field in the prisma schema, everytime a new record is created the field is automatically set to a UTC datetime, which is 1h off to the server time.

Here are some examples on how I tried to declare the field in the prisma schema.

```prisma
// simple declaration
changedOn  DateTime  @map("BEARBEITET_AM") @db.DateTime

// using the @updatedAt decorator
changedOn  DateTime  @updatedAt @map("BEARBEITET_AM") @db.DateTime

// using a default value and mapping it to the fields default restriction
changedOn  DateTime  @default(now(), map: "DF_MY_TABLE_BEARBEITET_AM") @map("BEARBEITET_AM") @db.DateTime
```
None of these set the correct time.

The only workaround I found was completely omitting the field from the schema. If I do that and create a new record the correct value for UTC+1 is set. But the result of that is that I can't read the field in my application, because it's not included in the response object of database request.

Any ideas or suggestions would be greatly appreciated.

Thank you.

### Information about Prisma Schema, Client Queries and Environment (optional)

- Client OS: Manjaro Linux 6.1.53
- Server OS: Windows Server 2019 Datacenter (10.0)
- Database: SQL Server 14.0.2052.1
- Node.js version: 18.18.0
- Prisma version: 5.3.1

**Comment by ludralph** [comment, marked as answer]
Hi @rstoye 👋 

Thank you for raising this question.

Prisma stores `DateTime` as UTC in the database, which explains the 1hr offset you are currently seeing. There are open [issues](https://github.com/prisma/prisma/issues/20615) to improve support for Timezones. A user in this previous [discussion](https://github.com/prisma/prisma/discussions/15329) shared a workaround. You can take a look to see if it helps resolve your issue.

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Reply by rstoye** [reply]
This helped understanding the issue better and the workaround also works for our case. Thank you.

**Reply by ludralph** [reply]
Thank you for confirming it helped for your use case. 🙌
